massa-consensus-worker/src/state/verifications.rs

```diff
@@ -113,7 +141,11 @@ impl ConsensusState {
         header: SecuredHeader,
         current_slot: Option<Slot>,
     ) -> Result<(), ConsensusError> {
-        match self.check_header(&block_id, &header, current_slot, self)? {
+        let header_outcome = self.check_header(&block_id, &header, current_slot, self)?;
+        if self.is_multistake(&header) {
+            return Ok(());
```

Comment:
problem: the check outcome is "Ok(HeaderCheckOutcome::Discard)" if the PoS draw is incorrect, so we should not add the header in the multistake cache in that outcome

Reply:
Actually we should only add it if it is ` HeaderCheckOutcome::Proceed`

Reply:
So we will ask for dependencies of a block that is potentially the 10000 we receive ?

Reply:
good point... then let's stick to everything but HeaderCheckOutcome::Discard

Reply:
Done